homeassistant/core.py

```diff
@@ -812,6 +814,22 @@ async def async_stop(self, exit_code: int = 0, *, force: bool = False) -> None:
         if self._stopped is not None:
             self._stopped.set()
 
+    def _cancel_cancellable_timers(self) -> None:
+        """Cancel timer handles marked as cancellable."""
+        # pylint: disable-next=[protected-access]
+        handles: Iterable[asyncio.TimerHandle] = self.loop._scheduled  # type: ignore[attr-defined]
+        for handle in handles:
+            if (
+                not handle.cancelled()
+                and (args := handle._args)  # pylint: disable=[protected-access]
+                and getattr(
```

Comment:
Maybe check the type is HassJob?

Reply:
I thought `getattr` would be cheaper than `isinstance`.

Reply:
It is but likely aren't going to subclass hassjob so we can do 

type(args[0]) is HassJob

Reply:
Done